omod/src/main/java/org/openmrs/module/rwandahivflowsheet/web/model/FormDataModel.java

```diff
@@ -432,21 +434,33 @@ public String getCurrentDrugNames(String drugSetName) {
 			if (o.getDrug() != null)
 				ret.append(o.getDrug().getName());
 			else
-				ret.append(o.getConcept().getBestName(Context.getLocale()).getName());
+				ret.append(o.getConcept().getName(Context.getLocale()).getName());
 			if (i.hasNext())
 				ret.append(" ");
 		}
 		return ret.toString();
 	}
 	
-	
+	public List<DrugOrder> getDrugOrdersByPatient(Patient patient){
+		List<DrugOrder> ret = new ArrayList<DrugOrder>();
+		List<Order> allPatientOrders = Context.getOrderService().getAllOrdersByPatient(patient);
+		if (allPatientOrders != null && allPatientOrders.size() > 0) {
+			for (Order order : allPatientOrders) {
+				if(order.getOrderType().equals(Context.getOrderService().getOrderTypeByUuid(OrderType.DRUG_ORDER_TYPE_UUID))) {
```

Comment:
Ideally you'd look this up once outside of the loop and not continuously look this up from the service.  Caching likely means it doesn't matter, but could be inefficient.

Reply:
I couldn't find a replacement for [getDrugOrdersByPatient](https://resources.openmrs.org/doc-1.9/org/openmrs/api/OrderService.html). The method has been letterly removed from core, and I was forced to get all orders for a patient and loop through all of them checking each one to see if it is actually a drug order.
Do you know a better method we can use here?

Reply:
[Here](https://github.com/PIH/openmrs-module-rwandahivflowsheet/pull/2/commits/49ca7831c7a2659a22130e2b8953c95addc2270b) is the commit after reading @dkayiwa feedback